File: 49-group-anagrams/group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        dic = defaultdict(list)
        for string in strs :
            """c = [0]*26
            for char in string:
                c[ord(char)-ord('a')] += 1
            dic[tuple(c)].append(string)"""
            sort_str = "".join(sorted(string))
            dic[sort_str].append(string)
        return  list(dic.values())

File: 49-group-anagrams/group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string,vector<string>> hash;
        for (string str : strs ){
            vector<int> freq(26,0);
            for (char c : str){ freq[c-'a']++;}
            string key = to_string(freq[0]);
            for (int i = 1; i < 26; i++){
                key += ',' + to_string(freq[i]);
            }
            hash[key].push_back(str);
        }
        vector<vector<string>> res;
        for (auto pair : hash){
            res.push_back(pair.second);
        }
        return res;
    }
};